src/domain/exchange/entities/exchange-result.entity.ts

```diff
@@ -1,6 +1,3 @@
 export interface ExchangeResult {
   success: boolean;
```

Comment:
I think having an interface with just `success` for two payloads is not required and might make some parts of the code more difficult to read (eg.: validators where we distinguish by type).

My suggestion would be to have this field set in `RatesExchangeResult` and `FiatCodesExchangeResult`.

Then we have a validator for `RatesExchangeResult` and one for `FiatCodesExchangeResult`. Wdyt?